test/functional/rpc_getorphantxs.py

```diff
@@ -125,6 +127,29 @@ def orphan_details_match(self, orphan, tx, verbosity):
             self.log.info("Check the transaction hex of orphan")
             assert_equal(orphan["hex"], tx["hex"])
 
+    def test_max_orphan_amount(self):
+        self.log.info("Check that once we reach the max orphan amount we dont get any new orphans")
+        node = self.nodes[0]
+
+        peer_1 = node.add_p2p_connection(P2PInterface())
+
+        self.log.info("Filling up orphanage with " + str(MAX_ORPHANS) + "(MAX_ORPHANS) orphans")
+        for i in range(MAX_ORPHANS):
+            tx_parent_1 = self.wallet.create_self_transfer()
+            tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
+            peer_1.send_and_ping(msg_tx(tx_child_1["tx"]))
```

Comment:
The test is currently a bit slow, and I think it might be because you're using `send_and_ping`, which waits for the other peer to send back a pong in each iteration of the loop. 

You could send the messages in the for loop and then ping only at the end:
```python
        for i in range(MAX_ORPHANS):
            tx_parent_1 = self.wallet.create_self_transfer()
            tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
            peer_1.send_message(msg_tx(tx_child_1["tx"]))

        peer_1.sync_with_ping()

        orphanage = node.getorphantxs()
        assert_equal(len(orphanage), MAX_ORPHANS)
```

The drawback would be that you're not checking that every single tx is entering the orphanage, although you could do that simply by saving the `orphans` and checking after the ping:

```python
        orphans = []
        for i in range(MAX_ORPHANS):
            tx_parent_1 = self.wallet.create_self_transfer()
            tx_child_1 = self.wallet.create_self_transfer(utxo_to_spend=tx_parent_1["new_utxo"])
            orphans.append(tx_child_1["tx"])
            peer_1.send_message(msg_tx(tx_child_1["tx"]))

        peer_1.sync_with_ping()

        orphanage = node.getorphantxs()
        assert_equal(len(orphanage), MAX_ORPHANS)

        for orphan in orphans:
            assert tx_in_orphanage(node, orphan)
```
Personally I think the first option is good enough though, what do you think?

Reply:
Thank you for the review! I've updated in [4d59b9a](https://github.com/bitcoin/bitcoin/pull/31040/commits/4d59b9a517afb5eb661b7bc179c51141769658a3)

the second one looked good to me! I prefer asserting that each individual tx made it into the orphanage